lib/model/rwfolder.go

```diff
@@ -1811,13 +1811,20 @@ func (f *sendReceiveFolder) deleteDir(dir string, ignores *ignore.Matcher, scanC
 	return err
 }
 
-// A []fileError is sent as part of an event and will be JSON serialized.
-type fileError struct {
+func (f *sendReceiveFolder) PullErrors() []FileError {
+	if state, _, _ := f.getState(); state == FolderSyncing {
+		return nil
```

Comment:
Why do we care about the state here?

Reply:
If the folder is syncing, the error list has just been cleared, is building up but not complete. Once it is, it will be sent on the folder error event. It's kind of similar as in the UI, while sync is in progress, there are no errors. But as you commented before, that premise is probably not ideal. Maybe you can comment on the last paragraph in https://github.com/syncthing/syncthing/pull/4650#issuecomment-355743842 , that would be the way I would go forward to change how failed items work (I didn't really understand your earlier proposition except for clearing errors after pulling).